reports.go

```diff
@@ -105,3 +825,56 @@ func (c *Client) CreateReport(reqPayload *CreateReportRequest) (*Report, error)
 
 	return createResp, nil
 }
+
+const getReport = "/v1/reports"
+
+// GetReport Retrieves a report
+func (c *Client) GetReport(reportID string) (*Report, error) {
+	rel, err := url.Parse(getReport)
+	if err != nil {
+		return nil, err
+	}
+
+	u := *c.BaseURL
+	u.Path = path.Join(u.Path, rel.String(), reportID)
+
+	req, err := http.NewRequest(http.MethodGet, u.String(), nil)
+	if err != nil {
+		return nil, err
+	}
+	req.Header.Add("Content-Type", "application/json")
+	req.SetBasicAuth(c.APIKey, "")
+
+	resp, err := c.client.Do(req)
+	if err != nil {
+		return nil, err
+	}
+
+	if resp.StatusCode == http.StatusNotFound {
+		return nil, nil
+	}
+
+	if resp.StatusCode != http.StatusOK {
+		return nil, NewResponseError([]int{
+			http.StatusCreated,
```

Comment:
@chandradeepak the status code here should be the `expected` status code. So it would be `http.StatusOK`.